### PR TITLE
Add ArrayStack

### PR DESCRIPTION
**Pull request by foikila**

### CODE REVIEW

**Comment by AdiChat** [comment]
As this consists of multiple files, create a folder `cpp` and add all files in it.

**Comment by foikila** [comment]
@AdiChat sure.

File: code/data_structures/abstract-data-structures/stack/IStack.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#ifndef ISTACK_H
```
Comment (AdiChat) [inline code/data_structures/abstract-data-structures/stack/IStack.h]: This shall be removed as this also exists in the `cpp` folder.

Comment (foikila) [reply code/data_structures/abstract-data-structures/stack/IStack.h]: Im thinking of adding an other implementation of the IStack -> LinkedStack. Still want me to move it?

Comment (AdiChat) [reply code/data_structures/abstract-data-structures/stack/IStack.h]: Okay. Looks good. 👍

**Comment by AdiChat** [comment]
Looks good 👍 

Hope you are enjoying your journey with **OpenGenus**

**Pull request merged**